### DISCUSSION TITLE
How to check the operation of "tmp_hlq"

### DISCUSSION THREAD

**Discussion by HanaOkubo** [ansible-collections/ibm_zos_core#765] [Q&A] [answered]
I would like to check the operation of "tmp_hlq". 
How can I do this with the following two modules?
・zos_fetch
・zos_dataset

I was able to confirm the other modules with backup names, but I don't know how to do it for the temporary data sets.
Also, in what situations is it useful to use tmp_hlq for temporary data sets?

**Comment by ketankelkar** [comment, marked as answer]
To address your second question first, it is useful to use `tmp_hlq` for temporary data sets in cases where the ansible user does not have sufficient permissions on the target z/OS system to create data sets using the default HLQ (high level qualifier) typically defined in our collection as the ansible user's username. For example, I run playbooks against my z/VM with the `ansible_user` set to `omvsadm` and thus the temporary HLQ used by the ibm_zos_core collection is `omvsadm` unless I specifically override via the `tmp_hlq` module parameter.

It is a bit tricky to visually confirm the temporary data sets which are created by the collection, largely because the `tmp_hlq` value is passed down to the ZOAU level. At that ZOAU level, the information is available if run in `verbose` mode but that output is not currently surfaced at the Ansible level (although that can be changed in the future). One potential option would be to make an equivalent call at the ZOAU level with verbosity turned on, however this would likely require reconstructing the command to ZOAU and would not necessarily confirm that Ansible follows the same behavior. Alternatively, it might be worth SSH-ing into the z/OS system and running an infinite loop using the ZOAU utility `dls` to constantly monitor for those temp data sets, something like:
```
while true; do dls HLQ.DATA.SET ; sleep 1; done
```
It looks like for `zos_fetch` the temp data set is created when a VSAM data set is fetched (i.e. `src` is a VSAM data set).
It looks like for `zos_data_set` the temp data set is created in the case of a PDS member delete operation.
